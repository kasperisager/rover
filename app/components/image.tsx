import React, { FunctionComponent } from "react";

import styles from "./image.module.css";

export const Image: FunctionComponent<Image.Properties> = ({
  children,
  src,
}) => {
  return (
    <div className={styles.image} style={{ backgroundImage: `url(${src})` }}>
      {children ? (
        <div className={styles.caption} tabIndex={0}>
          {children}
        </div>
      ) : null}
    </div>
  );
};

export namespace Image {
  export interface Properties {
    /**
     * The image source.
     */
    src: string;
  }
}
