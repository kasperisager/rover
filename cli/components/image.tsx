import React, { FunctionComponent } from "react";
import { Text } from "ink";

export const Image: FunctionComponent<Image.Properties> = ({ src }) => {
  return <Text>{src}</Text>;
};

export namespace Image {
  export type Properties = {
    src: string;
  };
}
