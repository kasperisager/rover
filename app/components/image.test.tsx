import React from "react";
import { create } from "react-test-renderer";

import { Image } from "./image";

test("<Image> renders with a src attribute", () => {
  const component = create(<Image src="https://example.com/foo.png" />);

  expect(component.toJSON()).toMatchSnapshot();
});

test("<Image> renders with children", () => {
  const component = create(
    <Image src="https://example.com/foo.png">Hello world</Image>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
