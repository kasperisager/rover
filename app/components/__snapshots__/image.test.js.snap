// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image> renders with a src attribute 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(https://example.com/foo.png)",
    }
  }
/>
`;

exports[`<Image> renders with children 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(https://example.com/foo.png)",
    }
  }
>
  <div
    tabIndex={0}
  >
    Hello world
  </div>
</div>
`;
